op-interop-mon/monitor/maintainer.go

```diff
@@ -11,12 +11,12 @@ import (
 )
 
 type Maintainer struct {
-	clients     locks.RWMap[eth.ChainID, *sources.EthClient]
-	finders     locks.RWMap[eth.ChainID, Finder]
-	updaters    locks.RWMap[eth.ChainID, Updater]
-	newInbox    chan *Job
-	updateInbox chan *Job
-	closed      chan struct{}
+	clients  locks.RWMap[eth.ChainID, *sources.EthClient]
+	finders  locks.RWMap[eth.ChainID, Finder]
+	updaters locks.RWMap[eth.ChainID, Updater]
```

Comment:
If I'm not mistaken, these three maps are actually only ever used by the (singleton) maintainer. So I don't think they need to be concurrency safe?

Reply:
Ah but sometimes we pass callbacks which invoke these to e.g. finders which may run concurrently. So they do need to be. 

I'm interested in a pattern where we don't use any callbacks at all and only use channels to wire everything up. To share memory by communicating (and not the other way around). So:

* finders send jobs (and notices about block finalization) to the maintainer, which "routes" or "forwards" them to updaters. The only state the maintainer  needs is a list of updaters.
* updaters receive jobs and finalization notices, and then spawns, updates and deletes jobs as appropriate
* the maintainer periodically scrapes jobs from updaters in order to update metrics (I wonder if this can also be improved, but that's another convo).

Reply:
No I think you're right, the way these maps are arranged makes them pretty static - they're all set up during resource init, and then it's all just reads. The Finders and Updaters have internals that are changing, but the references to them won't change.

This goes well with your other comment, the Maintainer can be significantly minimized I think. I may do that as part of this PR